sql/core/src/main/scala/org/apache/spark/sql/execution/python/ArrowPythonRunner.scala

```diff
@@ -126,18 +121,14 @@ class ArrowPythonRunner(
       private var schema: StructType = _
       private var vectors: Array[ColumnVector] = _
 
-      private var closed = false
-
       context.addTaskCompletionListener { _ =>
         // todo: we need something like `reader.end()`, which release all the resources, but leave
```

Comment:
@ueshin is this still a valid todo?  It is possible in Arrow now by calling `reader.close(false)` where the `false` flag will free resources but keep the stream open.

Reply:
Awesome. Then we should use the API at the end of arrow stream and here just in case.

Reply:
ok done